src/helpers/assertStringsWithinText.ts

```diff
@@ -0,0 +1,34 @@
+import { ReactTestRendererNode } from 'react-test-renderer';
+
+export const assertStringsWithinText = (
```

Comment:
The current structure consists of two recursive functions calling both themselves and each other.

The issue here results probably from the fact that there is no `parent` property on `toJSON` outputed nodes (unlike `ReactTestInstances`), as a workaround we could pass both parent and child to assertion function:

```ts
export function assertStringsWithinText(root: ReactTestRendererNode | Array<ReactTestRendererNode> | null) {
  if (!root) return;

  if (Array.isArray(root)) {
    root.forEach(child => _assertStringsWithinTextInternal(root, null));
    return;
  }

  _assertStringsWithinTextInternal(root, null);
}

function _assertStringsWithinTextInternal(node: ReactTestRendererNode, parent: ReactTestRendererNode) {
  if (typeof node === 'string' && parent.type !== "Text") {
    throw Error("Invariant violation: Text .... ");
  } 

  if (node.children) {
    node.children.forEach(child => _assertStringsWithinTextInternal(child, node));
  }
}
```

In the above setup only the internal function is recursive (to itself), the export functions servers as entry point to recursive one, but it only called once. 

Note that there is a slight difference in behaviour, as the above code would throw for root string value, which I think is reasonable, as that string value is not rendered inside `<Text>` component.

Wdyt?

Reply:
nit: BTW since we use `validate` in both option name `unstable_validateStringsRenderedInText` as well as `renderWithStringValidation` then maybe we should call this:

```
function validateStringsWithinText(...)
```

Wdyt?

Reply:
Oh good catch, actually it's just a mistake, the second function is intented to call only itself I just mixed up and didn't notice since it worked. Calling directly assertStringsWithinTextForNode instead of assertStringsWithinText will make no difference since calling assertStringsWithinText with a ReactTestRendererNode calls  assertStringsWithinTextForNode, so I just made the replacement